node/src/components/networking_metrics.rs

```diff
@@ -0,0 +1,66 @@
+use prometheus::{IntCounter, IntGauge, Registry};
+
+/// Network-type agnostic networking metrics.
+pub(crate) struct NetworkingMetrics {
+    /// How often a request was made a component to broadcast.
+    pub(crate) broadcast_requests: IntCounter,
+    /// How often a request to send a message directly to a peer was made.
+    pub(crate) direct_message_requests: IntCounter,
+    /// Current number of open connections.
+    pub(crate) open_connections: IntGauge,
+    /// Number of messages still waiting to be sent out (broadcast and direct).
+    pub(crate) queued_messages: IntGauge,
+
+    /// Registry instance.
+    registry: Registry,
+}
+
+impl NetworkingMetrics {
+    /// Creates a new instance of networking metrics.
+    pub fn new(registry: &Registry) -> Result<Self, prometheus::Error> {
+        let broadcast_requests =
+            IntCounter::new("net_broadcast_requests", "number of broadcasting requests")?;
+        let direct_message_requests = IntCounter::new(
+            "net_direct_message_requests",
+            "How often a request to send a message directly to a peer was made.",
```

Comment:
Maybe for consistency with `broadcast_requests`:
```suggestion
            "number of requests to send a message directly to a peer",
```
(Also for the other ones.)